src/lug/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1,12 +1,203 @@
+#include <cstdlib>
 #include <lug/Window/Win32/WindowImplWin32.hpp>
 
-bool lug::Window::priv::WindowImpl::create(uint16_t /*width*/, uint16_t /*height*/, const std::string & /*title*/, Style /*style*/) {
-    return false;
+lug::Window::priv::WindowImpl::WindowImpl(Window* win) : _parent(win) {
+
 }
 
-void lug::Window::priv::WindowImpl::close() {
+bool lug::Window::priv::WindowImpl::create(const std::string& title, Style style) {
     
+    // Register the window class at first call
+    if (_windowCount == 0) {
+        registerWindow();
+    }
+
+    int left;
+    int top;
+    // Compute position and size
+    {
+        HDC screenDC = GetDC(NULL);
+        left = (GetDeviceCaps(screenDC, HORZRES) - static_cast<int>(_parent->_mode.width)) / 2;
+        top = (GetDeviceCaps(screenDC, VERTRES) - static_cast<int>(_parent->_mode.height)) / 2;
+        ReleaseDC(NULL, screenDC);
+    }
+
+    // Choose the window style according to the Style parameter
+    DWORD win32Style = WS_VISIBLE;
+    if (style == Style::None) {
+        win32Style |= WS_POPUP;
+    } else {
+        if (static_cast<bool>(style & Style::Titlebar))
+        {
+            win32Style |= WS_CAPTION | WS_MINIMIZEBOX;
+        }
+        if (static_cast<bool>(style & Style::Resize))
+        {
+            win32Style |= WS_THICKFRAME | WS_MAXIMIZEBOX;
+        }
+        if (static_cast<bool>(style & Style::Close))
+        {
+            win32Style |= WS_SYSMENU;
+        }
+    }
+
+    _fullscreen = static_cast<bool>(style & Style::Fullscreen);
+    uint16_t realWidth;
+    uint16_t realHeight;
+    // In windowed mode, adjust width and height so that window will have the requested client area
+    if (!_fullscreen)
+    {
+        RECT rectangle = { 0, 0, _parent->_mode.width, _parent->_mode.height };
+        AdjustWindowRect(&rectangle, win32Style, false);
+        realWidth = rectangle.right - rectangle.left;
+        realHeight = rectangle.bottom - rectangle.top;
+    }
+
+    // Create the window
+    {
+        std::vector<wchar_t> realTitle(title.length() + 1); // Convert a const char * to a wchar_t
+        mbstowcs(&realTitle[0], title.c_str(), title.length() + 1);
+
+        _handle = CreateWindowW(_className, &realTitle[0], win32Style, left, top, realWidth, realHeight, NULL, NULL, GetModuleHandle(NULL), this);
+    }
+
+    // Switch to fullscreen if requested
+    if (_fullscreen)
+    {
+        setFullscreen();
```

Comment:
I think that the name of this function isn't appropriate, because "set" means that you can just set it to `true` or `false`.